prov/efa/src/efa_env.c

```diff
@@ -163,7 +162,14 @@ void efa_env_param_get(void)
 			    &efa_env.efa_read_segment_size);
 	fi_param_get_size_t(&efa_prov, "inter_max_gdrcopy_message_size",
 			    &efa_env.efa_max_gdrcopy_msg_size);
-	fi_param_get_bool(&efa_prov, "use_sm2", &efa_env.use_sm2);
+	fi_param_get_str(&efa_prov, "intranode_provider", &efa_env.intranode_provider);
+	fi_param_get_int(&efa_prov, "enable_shm_transfer", &efa_env.enable_shm_transfer);
+	if (efa_env.enable_shm_transfer == 0) {
+		efa_env.intranode_provider = "efa";
+		EFA_WARN(FI_LOG_CORE, "FI_EFA_ENABLE_SHM_PROVIDER is deprecated (and will be removed in a future release), "
```

Comment:
`FI_EFA_ENABLE_SHM_PROVIDER` -> `FI_EFA_ENABLE_SHM_TRANSFER`

Reply:
I think fi_param_get_str(&efa_prov, "intranode_provider", &efa_env.intranode_provider); whould be moved to after L172, 

Also, if `efa_env.intranode_provider` is none of `efa`, `shm`, sm2`, we should fail

Reply:
BY fail... we shouldn't abort, we should warn and run with EFA... we decided before that we want this behavior.